### PR TITLE
feat: host volume creation

### PR DESCRIPTION
**Pull request by attachmentgenie**
added defined type to be able to pass a hash of them to the nomad class and have them managed at the same time, this will give the option to switch to something like lvm easier (establish interface) and allows folks just to add a line to the hiera yaml

### CODE REVIEW

**Review by ischeinkman — APPROVED** [review]
While I don't know Puppet, I've left a rubber stamp and some questions to help me better understand puppet as a whole.

File: manifests/scheduling/nomad/host_volume.pp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  Enum['absent','directory'] $ensure = 'directory',
+  Stdlib::Filemode $mode = '0777',
+) {
+  file { $name:
+    ensure => $ensure,
+    force  => true,
+    mode   => $mode,
+    path   => "${data_path}/${name}",
+  }
+}
```
[the remark this answers is not in the document]
Comment (attachmentgenie) [reply manifests/scheduling/nomad/host_volume.pp:19]: it does indeed

File: manifests/scheduling/nomad.pp [hunk 2]
```diff
@@ -70,12 +71,19 @@
...
+  create_resources( '::profiles::scheduling::nomad::host_volume', $host_volumes, $host_volume_defaults)
+
   if $manage_disk {
     ::profiles::bootstrap::disk::mount { 'nomad data disk':
       device    => $device,
       mountpath => $data_path,
       before    => Service['nomad'],
     }
+
+    Profiles::Bootstrap::Disk::Mount <||> -> Profiles::Scheduling::Nomad::Host_volume <||>
```
[the remark this answers is not in the document]
Comment (attachmentgenie) [reply manifests/scheduling/nomad.pp:86]: This makes sure the mount is created before the directory. Puppet resources are not created top to bottom as one might expect. They are allowed to be created at random. This is a deliberate design decision on their part, this way you need to be explicit if you need there to be a order of business. The <[]> thing just denotes all

[the remark this answers is not in the document]
Comment (attachmentgenie) [reply manifests/scheduling/nomad.pp:86]: come to think of it i can make the first half more specific, will fix that in a follow up PR

[the remark this answers is not in the document]
Comment (ischeinkman) [reply manifests/scheduling/nomad.pp:86]: Thanks!

File: manifests/scheduling/nomad/host_volume.pp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ischeinkman) [reply manifests/scheduling/nomad/host_volume.pp:19]: Thanks for confirming!

**Pull request merged**